server/neptune/temporalworker/server.go

```diff
@@ -0,0 +1,192 @@
+	}
+}
+
+type Server struct {
+	Logger           logging.Logger
+	SSLCertFile      string
+	SSLKeyFile       string
+	Negroni          *negroni.Negroni
+	LyftMode         server.Mode
+	Port             int
+	StatsScope       tally.Scope
+	StatsCloser      io.Closer
+	TemporalHostPort string
+	JobsController   *controllers.JobsController
+}
+
+// TODO: as more behavior is added into the TemporalWorker package, inject corresponding dependencies
+func NewServer(config *Config) (*Server, error) {
+	jobsController := &controllers.JobsController{
+		StatsScope: config.Scope,
+	}
+	// router initialization
+	router := mux.NewRouter()
+	router.HandleFunc("/healthz", Healthz).Methods("GET")
+	router.HandleFunc("/jobs/{job-id}", jobsController.GetProjectJobs).Methods("GET").Name(ProjectJobsViewRouteName)
+	router.HandleFunc("/jobs/{job-id}/ws", jobsController.GetProjectJobsWS).Methods("GET")
+	n := negroni.New(&negroni.Recovery{
+		Logger:     log.New(os.Stdout, "", log.LstdFlags),
+		PrintStack: false,
+		StackAll:   false,
+		StackSize:  1024 * 8,
+	})
+	n.UseHandler(router)
+	server := Server{
+		TemporalHostPort: config.TemporalHostPort,
+		Logger:           config.CtxLogger,
+		SSLCertFile:      config.SslCertFile,
+		SSLKeyFile:       config.SslKeyFile,
+		StatsScope:       config.Scope,
+		StatsCloser:      config.Closer,
+		Negroni:          n,
+		JobsController:   jobsController,
+	}
+	return &server, nil
+}
+
+func (s Server) Start() error {
+	defer s.Logger.Close()
+
+	// temporal client + worker initialization
+	temporalClient, err := s.buildTemporalClient()
+	if err != nil {
+		return err
+	}
+	defer temporalClient.Close()
+	go func() {
+		w := worker.New(temporalClient, DeployTaskqueue, worker.Options{
+			// ensures that sessions are preserved on the same worker
+			EnableSessionWorker: true,
+		})
+		if err := w.Run(worker.InterruptCh()); err != nil {
+			log.Fatalln("unable to start worker", err)
+		}
+	}()
+
+	// Ensure server gracefully drains connections when stopped.
+	stop := make(chan os.Signal, 1)
+	// Stop on SIGINTs and SIGTERMs.
+	signal.Notify(stop, os.Interrupt, syscall.SIGTERM)
+
+	s.Logger.Info(fmt.Sprintf("Atlantis started - listening on port %v", s.Port))
+	httpServerProxy := httpServerProxy{
+		SSLCertFile: s.SSLCertFile,
+		SSLKeyFile:  s.SSLKeyFile,
+		Server:      &http.Server{Addr: fmt.Sprintf(":%d", s.Port), Handler: s.Negroni},
+		Logger:      s.Logger,
+	}
+	go httpServerProxy.ListenAndServe()
+	<-stop
+
+	// flush stats before shutdown
+	if err := s.StatsCloser.Close(); err != nil {
+		s.Logger.Error(err.Error())
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+	if err := httpServerProxy.Shutdown(ctx); err != nil {
+		return cli.NewExitError(fmt.Sprintf("while shutting down: %s", err), 1)
+	}
+	return nil
+}
+
+func (s *Server) buildTemporalClient() (client.Client, error) {
+	if s.TemporalHostPort != "" {
+		return nil, errors.New("invalid host port")
+	}
```

Comment:
Is this possible that this would be empty? Should we just be validating this in the global cfg parsing instead of having this check in multiple zones?

Reply:
i'll revert back to an add if not empty check. client options already provides a default if not set.